compiler/executable/match_/planner/plan.rs

```diff
@@ -941,29 +941,42 @@ impl PartialCostPlan {
     }
 
     fn determine_joinability(&self, graph: &Graph<'_>, pattern: PatternVertexId) -> Option<VariableVertexId> {
-        let mut updated_join_var: Option<VariableVertexId> = None;
-        if let Some(prev_constraint) = self.ongoing_step.iter().next() {
+        if let Some(&prev_pattern) = self.ongoing_step.iter().next() {
```

Comment:
This could be short-circuited rather than wrapping the whole of the function body in an if-let.
```suggestion
       let &prev_pattern = self.ongoing_step.iter().next()?;
```